core/src/main/java/com/linecorp/armeria/client/ClientRequestContextBuilder.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client;
+
+import static java.util.Objects.requireNonNull;
+
+import java.net.URI;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.AbstractRequestContextBuilder;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.RpcRequest;
+
+/**
+ * Builds a new {@link ClientRequestContext}. Note that it is not usually required to create a new context by
+ * yourself, because Armeria will always provide a context object for you. However, it may be useful in some
+ * cases such as unit testing.
+ */
+public final class ClientRequestContextBuilder
+        extends AbstractRequestContextBuilder<ClientRequestContextBuilder> {
+
+    /**
+     * Returns a new {@link ClientRequestContextBuilder} created from the specified {@link HttpRequest}.
+     */
+    public static ClientRequestContextBuilder of(HttpRequest request) {
+        return new ClientRequestContextBuilder(request);
+    }
+
+    /**
+     * Returns a new {@link ClientRequestContextBuilder} created from the specified {@link RpcRequest} and URI.
+     */
+    public static ClientRequestContextBuilder of(RpcRequest request, String uri) {
```

Comment:
Why does `ClientRequestContextBuilder` accept `RpcRequest` but not `ServiceRequestContextBuilder`? If the use case is for a server that processes messages from Kafka, for example, it seems like a service (the client is whoever published the message).

Reply:
Because a `ServiceRequestContext`'s request has been always an `HttpRequest`? For instance, `ServiceRequestContext.service()` always returns `Service<HttpRequest, HttpResponse>`. We may want to change this if we support more than HTTP but I guess it would be hard to occur at the moment.

Reply:
In that case what would it look like to model the kafka request with `ServiceRequestContext`? Would it need a dummy HTTP request? I'm trying to understand how this will solve the use case, at first I expected to create a `ServiceRequestContext` with `RpcRequest.of(KafkaService.class` or something